### PR TITLE
Input: Allow browser UI for picking date via date component

### LINKED ISSUE #3235: [ENHANCEMENT] Allow browser UI for picking date via date component
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement

The Kirby input-field of type date should support selecting a date using the browsers native date picker. 

## Describe the solution you'd like

The Kirby input-field should be able to open a date picker for selecting a date.

- [ ] The date picker should use the browsers native implementation

## Have you considered any alternatives?

Later on a custom date picker (using Kirby `Calendar` component) can be created and replace the browsers default, but to have a MVP the date picker should simply be the browsers default.

## Are there any additional context?
In issue #3234 a calendar icon will be added. Clicking this icon should open the date picker.

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by jkaltoft**
## Which issue does this PR close?

This PR closes #3235

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/form-field/src/_form-field-inputs.shared.scss [hunk 1]
```diff
@@ -69,3 +75,12 @@ $form-field-label-height: 24px;
     color: var(--kirby-inputs-placeholder-color);
   }
 }
+
+// Firefox, Safari, Mobile Safari
+@supports not (selector(::-webkit-date-and-time-value)) {
+  // WTF? 😄
```
Comment (jakobe) [inline libs/designsystem/form-field/src/_form-field-inputs.shared.scss]: 😂

File: libs/designsystem/form-field/src/_form-field-inputs.shared.scss [hunk 2]
```diff
@@ -69,3 +75,12 @@ $form-field-label-height: 24px;
...
+  :host([type='date']::-webkit-date-and-time-value) {
+    // Enforce consistent height in Mobile Safari
+    min-height: #{$form-field-input-line-height + em};
```
Comment (jakobe) [inline libs/designsystem/form-field/src/_form-field-inputs.shared.scss:86]: Is this related to the issue at hand or an unrelated enhancement? 🤔

**Review by jakobe — COMMENTED** [review]
Good job 🙌🏻 

Just a single comment...

File: libs/designsystem/form-field/src/_form-field-inputs.shared.scss (hunk 1 above)
Comment (jkaltoft) [reply libs/designsystem/form-field/src/_form-field-inputs.shared.scss]: yup. weird shit 😂

I'm considering adding a more descriptive comment, though

File: libs/designsystem/form-field/src/_form-field-inputs.shared.scss (hunk 2 above)
Comment (jkaltoft) [reply libs/designsystem/form-field/src/_form-field-inputs.shared.scss:86]: Related. 

Without (conditional) `min-height` declaration:

image

With (conditional) `min-height` declaration:

image

Comment (jkaltoft) [reply libs/designsystem/form-field/src/_form-field-inputs.shared.scss:86]: 👆🏻

Only Mobile Safari seems to be affected. It has less intrinsic height initially (when no date has been selected yet) and when the date has been reset.

Declaring a `min-height` fixes this.

File: libs/designsystem/form-field/src/_form-field-inputs.shared.scss (hunk 1 above)
Comment (jkaltoft) [reply libs/designsystem/form-field/src/_form-field-inputs.shared.scss]: How about this?:

```scss
// Firefox, Safari, Mobile Safari
@supports not (selector(::-webkit-date-and-time-value)) {
  // WTF? 😄 - Mobile Safari *does* apply this ⬇️ despite @supports not ... ⬆️
  :host([type='date']::-webkit-date-and-time-value) {
    // Mobile Safari has less intrinsic height initially (when no date has been
    // selected yet) and when the date has been reset.
    // Declaring this min-height ensures consistent height in Mobile Safari
    min-height: #{$form-field-input-line-height + em};
  }
}
```

Comment (jakobe) [reply libs/designsystem/form-field/src/_form-field-inputs.shared.scss]: Perfect - much better 👏🏻

**Review by jakobe — APPROVED** [review]

**Pull request merged**